client/state/selectors/should-show-masterbar-notifications.js

```diff
@@ -0,0 +1,15 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * Returns true if masterbar notifications should be shown.
+ *
+ * @param  {Object}  state - Global state tree
+ * @return {Boolean} true  - if notifications should be shown.
+ *
+ */
+export default function shouldShowMasterbarNotifications( state ) {
+	return get( state, 'ui.masterbar.showNotifications', false );
```

Comment:
even with a rename I think it's fine to leave this under `ui.masterbar`. with the selector we are at least free to change it later internally without affecting other things